distribution/push_v2.go

```diff
@@ -670,10 +668,9 @@ func numOfMatchingPathComponents(pth string, matchComponents []string) int {
 }
 
 func getPathComponents(path string) []string {
-	// make sure to add docker.io/ prefix to the path
-	named, err := reference.ParseNamed(path)
+	named, err := reference.ParseNormalizedNamed(path)
```

Comment:
It looks like `SourceRepository` is always normalized. The entries are created here:

```go
                        // Cache mapping from this layer's DiffID to the blobsum
                        if err := pd.v2MetadataService.TagAndAdd(diffID, pd.hmacKey, metadata.V2Metadata{
                                Digest:           err.Descriptor.Digest,
                                SourceRepository: pd.repoInfo.FullName(),
                        }); err != nil {
                                return distribution.Descriptor{}, xfer.DoNotRetry{Err: err}
                        }
```